applications/file-split-ftp/src/main/java/org/springframework/integration/samples/filesplit/Application.java

```diff
@@ -0,0 +1,229 @@
+ */
+
+package org.springframework.integration.samples.filesplit;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.apache.commons.net.ftp.FTPFile;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.mail.MailProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.file.Files;
+import org.springframework.integration.dsl.mail.Mail;
+import org.springframework.integration.file.FileHeaders;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.integration.ftp.session.DefaultFtpSessionFactory;
+import org.springframework.integration.http.config.EnableIntegrationGraphController;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.MessagingException;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+@SpringBootApplication
+@EnableIntegrationGraphController
+public class Application {
+
+	private static final String EMAIL_SUCCESS_SUFFIX = "emailSuccessSuffix";
+
+	public static void main(String[] args) {
+		SpringApplication.run(Application.class, args);
+	}
+
+	@Autowired
+	private MailProperties mailProperties;
+
+	/**
+	 * Poll for files, add an error channel, split into lines route the start/end markers
+	 * to {@link #markers()} and the lines to {@link #lines()}.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow fromFile() {
+		return IntegrationFlows.from(Files.inboundAdapter(new File("/tmp/in"))
+					.preventDuplicates(false)
+					.patternFilter("*.txt"), e -> e.poller(Pollers.fixedDelay(5000)))
+				.enrichHeaders(h -> h.header(MessageHeaders.ERROR_CHANNEL, "tfrErrors.input"))
+				.handle(Files.splitter(true, true))
+				.<Object, Class<?>>route(Object::getClass, m -> m
+						.channelMapping(FileSplitter.FileMarker.class, "markers.input")
+						.channelMapping(String.class, "lines.input"))
+				.get();
+	}
+
+	/**
+	 * Process lines; append (no flush) to the appropriate file.
+	 * @return the flow.
+	 */
+	@Bean
+	public IntegrationFlow lines() {
+		return f -> f.enrichHeaders(h -> h.headerExpression(
+						FileHeaders.FILENAME, "payload.substring(1, 4) + '.txt'", true))
```

Comment:
It worked fine for me with the file name expression. I've built on linux too. Let's see what our friend Travis thinks.